### PR TITLE
Add go vet to Travis

### PR DESCRIPTION
**Pull request by bugosiphionah**
For #375 

cc @jrperritt , @jtopjian

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -1,5 +1,7 @@
 language: go
 sudo: false
+before_script:
```
Comment (jtopjian) [inline .travis.yml]: How about moving this down so it's placed above `script`?

```yaml
before_script:
- go vet ./...
script:
- ./script/coverage
- ./script/format
```

Comment (bugosiphionah) [reply .travis.yml]: Yes I will work on cleaning those errors. Thanks

**Comment by jtopjian** [comment]
@bugosiphionah Let me know if you need a hand with this. I did a bunch of `vet` fixes a few months ago and some of them were tricky.

**Comment by bugosiphionah** [comment]
@jtopjian I think I need some guidance here. When I run go vet ./... it works on my end. This is the output with no errors whatosoever.

```
programming@programming:~/go/src/github.com/gophercloud/gophercloud$ go vet ./...
programming@programming:~/go/src/github.com/gophercloud/gophercloud$ 
```
But fails when I commit..if you have anything to point out as I also investigate, It will be helpful.

**Comment by jtopjian** [comment]
@bugosiphionah Yep, I am seeing the same thing.

The issue is that have been commits to the `master` branch since you've begun working on this. One commit introduces a new `vet` error. You'll need to "rebase" your branch with the master branch:

```git shell
$ git checkout master
$ git pull
$ git checkout issue_375
$ git rebase -i master
```

You'll be presented with a screen like this:

```
pick 9249e0f Add go vet to Travis
pick 90db222 Add go vet to Travis
pick e3b87cc Add go vet to Travis
pick 751dd55 Add go vet
pick 5d748fe Fix go vet errors
pick 9206738 Add go vet
pick 7465a91 Fix go vet errors
pick eae2239 Fix go vet errors
pick 654785d Fix go vet errors
```

Don't modify anything - just save and continue.

Once the rebase is done, you can run `go vet ./...` and you'll now see the error that Travis is reporting. If you look at `users/requests.go`, you'll see that there has been a new function added.

When you go to push to Github, you _must_ do a forceful push now because of the rebase:

```shell
$ git push -f ...
```

Also, quick note about `openstack/networking/v2/extensions/external/testing/results_test.go`: you'll want to change `networkCreateOpts` to `CreateOptsBuilder` and `networkUpdateOpts` to `UpdateOptsBuilder` -- you have this correct in `acceptance/openstack/networking/v2/extensions/extensions.go` :)

Let me know if you have any other questions.

**Comment by bugosiphionah** [comment]
Thanks this is very helpful :+1:

**Comment by bugosiphionah** [comment]
@jtopjian ready for review.

**Comment by jtopjian** [comment]
@bugosiphionah Nice work :)

This looks good to me.

@jrperritt any issues with adding this to Travis?

**Comment by jrperritt** [comment]
> any issues with adding this to Travis?

Nope; sounds good to me

**Pull request merged**